src/analyses/apron/apronPriv.apron.ml

```diff
@@ -342,6 +344,8 @@ struct
   let threadenter ask getg (st: apron_components_t): apron_components_t =
     {apr = getg (); priv = startstate ()}
 
+  let iter_sys_vars getg vq vf = () (* TODO: or report singleton global for any Global query? *)
```

Comment:
> TODO: or report singleton global for any Global query? 

One would indeed expect that `vf` is applied to all constraint system unknowns that somehow relate to `vq`, i.e. here to the single global unknown. No?

Reply:
Well, it's not so clear cut what should and should not be considered to be related to a global `g`, i.e. what does "somehow relate" mean.

In this case, since all globals are related under a single constraint unknown, restarting one (based on the result of this query) would mean restarting them all! Same goes for the region analysis, which currently also doesn't report it.

Not only that, suppose `g` is part of a context (within any analysis), should `Global g` additionally be able to return all unknowns of all functions of all contexts that contain it? That requires a whole lot of extra mappings to collect. Not to mention, it would require some additional mechanism to allow checking if an abstract context (depending on activated analysis and their states) contains `g`.

Therefore to avoid overly complicating things for now, I thought it'd be enough to relate unknowns which very directly relate to a particular global or node.

Reply:
>In this case, since all globals are related under a single constraint unknown, restarting one (based on the result of this query) would mean restarting them all! Same goes for the region analysis, which currently also doesn't report it.

That's true. In a hypothetical setting where we have incremental restarting for the  apron analysis, I would just assume that a user might be confused, when he triggers a re-analysis of a certain global and nothing happens. But maybe the decision here only really matters when #558 is fixed. 

>Not only that, suppose g is part of a context (within any analysis), should Global g additionally be able to return all unknowns of all functions of all contexts that contain it? That requires a whole lot of extra mappings to collect. Not to mention, it would require some additional mechanism to allow checking if an abstract context (depending on activated analysis and their states) contains g.

Yes, that would certainly lead to far. Although in a `earlyglobs` setting this is not such issue, but with privatization it again would be.

Reply:
> I would just assume that a user might be confused, when he triggers a re-analysis of a certain global and nothing happens.

This is just the `protection` Apron privatization though. Others, including our default `mutex-meet` have separate unknowns per global and those are returned by the varquery, so it shouldn't be too bad.

I suppose if we at some point want better control over how much things related to a global should be restarted, more arguments could be added to that varquery to choose how specific or general things may be returned. Kind of a precision knob.